core/src/main/java/gyro/core/diff/Diff.java

```diff
@@ -279,6 +285,21 @@ private Change newDelete(Diffable diffable) {
         return delete;
     }
 
+    private void setUsesCredentialFromState(Diffable diffable) {
+        DiffableScope scope = DiffableInternals.getScope(diffable);
+
+        Node node = scope.getStateNodes()
+            .stream()
+            .filter(o -> o instanceof DirectiveNode)
+            .filter(o -> ((DirectiveNode) o).getName().equals("uses-credentials"))
```

Comment:
This should be generic rather that directly mention a specific uses-credentials.